sync_diff_inspector/diff.go

```diff
@@ -122,21 +123,44 @@ func (df *Diff) init(cfg *Config) (err error) {
 
 // CreateDBConn creates db connections for source and target.
 func (df *Diff) CreateDBConn(cfg *Config) (err error) {
-	for _, source := range cfg.SourceDBCfg {
-		source.Conn, err = diff.CreateDB(df.ctx, source.DBConfig, cfg.CheckThreadCount)
-		if err != nil {
-			return errors.Errorf("create source db %s error %v", source.DBConfig.String(), err)
-		}
-		df.sourceDBs[source.InstanceID] = source
-	}
-
 	// create connection for target.
-	cfg.TargetDBCfg.Conn, err = diff.CreateDB(df.ctx, cfg.TargetDBCfg.DBConfig, cfg.CheckThreadCount)
+	cfg.TargetDBCfg.Conn, err = diff.CreateDB(df.ctx, cfg.TargetDBCfg.DBConfig, nil, cfg.CheckThreadCount)
 	if err != nil {
 		return errors.Errorf("create target db %s error %v", cfg.TargetDBCfg.DBConfig.String(), err)
 	}
 	df.targetDB = cfg.TargetDBCfg
 
+	targetTZOffset, err := dbutil.GetTimeZoneOffset(df.ctx, cfg.TargetDBCfg.Conn)
+	if err != nil {
+		return errors.Annotatef(err, "fetch target db %s time zone offset failed", cfg.TargetDBCfg.DBConfig.String())
+	}
+	vars := map[string]string{
+		"time_zone": url.QueryEscape(dbutil.FormatTimeZoneOffset(targetTZOffset)),
+	}
+
+	for _, source := range cfg.SourceDBCfg {
+		source.Conn, err = diff.CreateDB(df.ctx, source.DBConfig, nil, cfg.CheckThreadCount)
+		if err != nil {
+			return errors.Annotatef(err, "create source db %s failed", source.DBConfig.String())
+		}
+
+		sourceTZOffset, err := dbutil.GetTimeZoneOffset(df.ctx, cfg.TargetDBCfg.Conn)
```

Comment:
```suggestion
		sourceTZOffset, err := dbutil.GetTimeZoneOffset(df.ctx, source.Conn)
```

And we could directly specify timezone, not try-and-adjust? 🤔